### PR TITLE
add support for rails params

### PR DESCRIPTION
**Pull request by mathieujobin**

### CODE REVIEW

**Comment by mathieujobin** [comment]
this is because `ActionController::Parameters` no longer inherits `Hash`

see this deprecation notice

DEPRECATION WARNING: Method symbolize_keys is deprecated and will be removed in Rails 5.1, as `ActionController::Parameters` no longer inherits from hash. Using this deprecated behavior exposes potential security problems. If you continue to use this method you may be creating a security vulnerability in your app that can be exploited. Instead, consider using one of these documented methods which are not deprecated: http://api.rubyonrails.org/v5.0.0.rc1/classes/ActionController/Parameters.html

**Comment by mathieujobin** [comment]
@ethnext this is needed for Rails 5

File: lib/api_hammer/faraday/request_logger.rb [hunk 1]
```diff
@@ -79,8 +79,13 @@ def call(request_env)
               'activesupport_tagged_logging_tags' => @log_tags,
             }.reject{|k,v| v.nil? },
           }
-
-          json_data = JSON.generate(data)
+          begin
+            json_data = JSON.generate(data)
+          rescue Encoding::UndefinedConversionError => e
+            # force encoding on request data in case of a conversion error
+            data["request"].each {|k,v| v.force_encoding('UTF-8') if String == v.class }
```
Comment (notEthan) [inline lib/api_hammer/faraday/request_logger.rb]: `if v.is_a?(String)` is a better way to check class

Comment (notEthan) [inline lib/api_hammer/faraday/request_logger.rb]: I don't much like just setting all strings to UTF8, that is not necessarily correct

Comment (notEthan) [inline lib/api_hammer/faraday/request_logger.rb]: I fixed tihs on the rack request logger but I guess not on the faraday one. whoops.

Comment (notEthan) [inline lib/api_hammer/faraday/request_logger.rb]: https://github.com/notEthan/api_hammer/pull/35

**Comment by notEthan** [comment]
still do want the subparams hash / params check, though

**Comment by mathieujobin** [comment]
reopening... i will remove that 2nd commit, so you can merge the first one

**Comment by notEthan** [comment]
merged this but I shouldn't have; ActionController may not be defined here, so I unmerged it but not sure if this can be reopened

**Comment by notEthan** [comment]
maybe checking if respond_to?(:to_h) and to_h'ing that is a more generic solution

**Comment by mathieujobin** [comment]
i need this for access.

**Pull request merged**